src/parameters/MicrophysicsP3.jl

```diff
@@ -1,63 +1,418 @@
     μ_max::FT
-    "Water freeze temperature [K]"
-    T_freeze::FT
-    "ventillation factor a"
+end
+function SlopePowerLaw(td::CP.AbstractTOMLDict)
+    name_map = (;
+        :Heymsfield_mu_coeff1 => :a,
+        :Heymsfield_mu_coeff2 => :b,
+        :Heymsfield_mu_coeff3 => :c,
+        :Heymsfield_mu_cutoff => :μ_max,
+    )
+    params = CP.get_parameter_values(td, name_map, "CloudMicrophysics")
+    FT = CP.float_type(td)
+    return SlopePowerLaw{FT}(; params...)
+end
+
+"""
+    SlopeConstant{FT}
+
+Slope parameter μ as a constant:
+    μ(λ) = μ_const
+
+# Fields
+$(DocStringExtensions.FIELDS)
+
+!!! note "Calling this function"
+    This function is not typically called directly. Instead, see [`ParametersP3`](@ref)
+
+# Examples
+
+```julia-repl
+julia> import CloudMicrophysics.Parameters as CMP
+julia> import ClimaParams as CP
+julia> override_file = Dict("Heymsfield_mu_coeff1" => Dict("value" => 3.0))
+julia> toml_dict = CP.create_toml_dict(Float64; override_file)
+julia> CMP.SlopeConstant(toml_dict)
+SlopeConstant{Float64}
+└── μ = 3.0 [-]
+```
+"""
+Base.@kwdef struct SlopeConstant{FT} <: SlopeLaw{FT}
+    "Slope parameter μ [-]"
+    μ::FT
+end
+function SlopeConstant(td::CP.AbstractTOMLDict)
+    name_map = (; :Heymsfield_mu_coeff1 => :μ)
+    params = CP.get_parameter_values(td, name_map, "CloudMicrophysics")
+    FT = CP.float_type(td)
+    return SlopeConstant{FT}(; params...)
+end
+
+"""
+    VentilationSB2005{FT}
+
+Parameters for ventilation factor:
+
+```math
+F(r) = a_{vent} + b_{vent}  N_{Sc}^{1/3} N_{Re}(r)^{1/2}
+```
+
+From Seifert and Beheng 2005, doi: 10.1007/s00703-005-0112-4
+
+# Fields
+$(DocStringExtensions.FIELDS)
+
+!!! note "Calling this function"
+    This function is not typically called directly. Instead, see [`ParametersP3`](@ref)
+
+# Examples
+
+```julia-repl
+julia> import CloudMicrophysics.Parameters as CMP
+julia> import ClimaParams as CP
+julia> toml_dict = CP.create_toml_dict(Float64)
+julia> CMP.VentilationSB2005(toml_dict)
+VentilationSB2005{Float64}
+├── vent_a = 0.78 [-]
+└── vent_b = 0.308 [-]
+```
+
+"""
+Base.@kwdef struct VentilationSB2005{FT} <: ParametersType{FT}
+    "Ventilation factor a [-]"
     vent_a::FT
-    "ventillation factor b"
+    "Ventilation factor b [-]"
     vent_b::FT
 end
+function VentilationSB2005(td::CP.AbstractTOMLDict)
+    name_map = (;
+        :p3_ventillation_a => :vent_a, # TODO: fix typo in TOML
+        :p3_ventiallation_b => :vent_b,
+    )
+    params = CP.get_parameter_values(td, name_map, "CloudMicrophysics")
+    FT = CP.float_type(td)
+    return VentilationSB2005{FT}(; params...)
+end
+
+### ----------------------------- ###
+### --- TOP-LEVEL CONSTRUCTOR --- ###
+### ----------------------------- ###
+
+"""
+    ParametersP3{FT}
+
+Parameters for P3 bulk microphysics scheme.
+    
+From [MorrisonMilbrandt2015](@cite)
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct ParametersP3{FT, SLOPELAW <: SlopeLaw{FT}} <:
```

Comment:
I thought that for some GPU compilation reasons we had to specify all the types here. But I could be wrong

Reply:
I showed this to Dennis. We are specifying all (parameterized) types. The types not specified up here are not needed because they are concrete types, as opposed to abstract types.